packages/carbon-web-components/src/components/ui-shell/side-nav.ts

```diff
@@ -179,6 +168,11 @@ class CDSSideNav extends HostListenerMixin(LitElement) {
         forEach(headerItems, (item) => {
           item.setAttribute('tabindex', '-1');
         });
+        (
+          this.querySelector(
+            (this.constructor as typeof CDSSideNav).selectorNavItems
+          ) as HTMLElement
+        )?.focus();
```

Comment:
Moving the logic out of the event listener and into the update lifecycle ensures we focus _after_ the seletorNavItems are added to the DOM.